crates/core/transaction/src/plan.rs

```diff
@@ -48,6 +49,20 @@ pub struct TransactionPlan {
 }
 
 impl TransactionPlan {
+    /// Sort the actions by type, using the protobuf field number in the [`Action`].
+    pub fn sort_actions(&self, mut actions: Vec<Action>) -> Vec<Action> {
+        actions.sort_by_key(|action: &Action| action.variant_index());
+
+        actions
+    }
+
+    /// Sort the action plans by type, using the protobuf field number in the [`ActionPlan`].
+    pub fn sort_action_plans(&self, mut actions_plans: Vec<ActionPlan>) -> Vec<ActionPlan> {
+        actions_plans.sort_by_key(|action: &ActionPlan| action.variant_index());
+
+        actions_plans
+    }
```

Comment:
I think we probably just want to have a single method
```
TransactionPlan::sort(&mut self) { ... }
```
right? Then we can sort the plan exactly once, immediately before the planner returns it, and then we won't have to change any of the other code.

Reply:
yeah that's much cleaner